mlx/backend/common/cholesky.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright © 2023-2024 Apple Inc.
+
+#include "mlx/allocator.h"
+#include "mlx/backend/common/copy.h"
+#include "mlx/linalg.h"
+#include "mlx/primitives.h"
+
+#ifdef ACCELERATE_NEW_LAPACK
+#include <Accelerate/Accelerate.h>
+#else
+#include <lapack.h>
+#endif
+
+namespace mlx::core {
+
+void cholesky_impl(const array& a, array& T, bool upper) {
+  // Lapack uses the column-major convention. We take advantage of the fact that
+  // the matrix should be symmetric:
+  //   (A)ᵀ = A
+  // and that a column-major lower triangular matrix is a row-major upper
+  // triangular matrix, so uplo is the opposite of what we would expect from
+  // upper
+
+  char uplo;
+  if (upper) {
+    uplo = 'L';
+  } else {
+    uplo = 'U';
+  }
+
+  // The decomposition is computed in place, so just copy the input to the
+  // output.
+  copy(a, T, a.flags().row_contiguous ? CopyType::Vector : CopyType::General);
```

Comment:
That's great and should also handle donation. Very nice.

Reply:
If I understood correctly the donation, yes! It's done through ```CopyType::Vector```